### ISSUE TITLE
Remove obsolete Main*.cpp files (and corresponding executables) from Albany/src directory? 

### ISSUE THREAD

**Issue by ikalash** [SNLComputation/Albany#456]
When I was working on issue #453 , I realized we have a lot of obsolete Main*.cpp files in the Albany/src directory.  The SG ones:

Main_SGAdjoint.cpp
Main_SGCoupled.cpp 
Main_SGSolve.cpp

are not compiled, and we don't have SG anymore, so I think it's pretty safe to remove those.  

I wrote  Main_SolveT_Tempus.cpp when I was first hooking up Albany to Tempus.  This executable was for testing, and it hooks up Tempus directly to Albany w/o going through Piro.  It's capabilities were limited and it was for debugging only.  Therefore it can be removed.

Main_Alternating.cpp was also an experiment by @lxmota that was ultimately dropped, so I think it can be removed (Alejandro, please chime in if I am wrong on this).  

The executables from 

Main_Adjoint.cpp
Main_SolveT_MPMD.cpp 

are not used by any tests.  I can see from CMakeLists.txt that Main_SolveT_MPMD.cpp is supposed to be for ATO but it is not tests - @jrobbin can you please comment on whether this is used / needed?  

To summarize, it seems the relevant Main files are:

Main_Analysis.cpp
Main_AnalysisT.cpp
Main_Dakota.cpp 
Main_DakotaT.cpp 
Main_Solve.cpp
Main_SolveT.cpp
Potentially Main_SolveT_MPMD.cpp depending on what @jrobbin says (though since it has not been tested likely in years I am wary about it)

Any others people want to keep?  Anyone opposed to removing the obsolete Main files?

**Comment by ikalash** [comment]
I forgot Main_Coupled.cpp.  This is used by CoupledReactor and NSReactor created by @etphipp according to git blame.  I do remember this Main file being very useful for our Schwarz work as an example.  I would vote to keep it on these grounds + the fact that unlike some of the other executables, it is tested.

**Comment by ikalash** [comment]
To follow up, I've removed all the mains except: 

Main_Analysis.cpp
Main_AnalysisT.cpp
Main_Dakota.cpp
Main_DakotaT.cpp
Main_Solve.cpp
Main_SolveT.cpp
Main_SolveT_MPMD.cpp (waiting to hear back from @jrobbin about whether this one is needed in master)

The Dakota tests are no longer run in any of the nightlies.  I used to run them on my machine but I turned them off some time ago, I believe because there was some TriKota compilation error that never got resolved that was messing up the builds.  If we wish to keep the Dakota mains / executables, we should fix the issues there and turn it on in at least one nightly.

**Comment by ikalash** [comment]
I have checked with @jrobbin and the Main_SolveT_MPMD.cpp file is important, and is tested as a part of PLATO.  Therefore it is to be kept.  I've removed all the unneeded Mains in various commits, therefore closing this issue.
